geth/api/backend.go

```diff
@@ -34,12 +40,14 @@ func NewStatusBackend() *StatusBackend {
 	accountManager := account.NewManager(nodeManager)
 	txQueueManager := txqueue.NewManager(nodeManager, accountManager)
 	jailManager := jail.New(nodeManager)
+	notificationManager := fcm.NewNotifier(fcmServerKey)
```

Comment:
Wouldn't a separate method be simpler than making a function returning function accessing a previously passed argument as in a closure?

Reply:
I dont think so, higher order function looks cleaner and more safe, isnt it?

Reply:
Come on :)

Reply:
And what about this? Obviously, higher-order functions only adds complexity, I don't see why we need it here instead of just using a constructor, can you explain, please?

Reply:
With higher order function:
```
//Declaration
notifier := fcm.NewNotifier(fcmServerKey)

// Usage
err := api.b.notifier().Send(msg, token)
```

Without:
```
diff --git a/geth/api/api.go b/geth/api/api.go
index 170faf40..d519cda7 100644
--- a/geth/api/api.go
+++ b/geth/api/api.go
@@ -7,6 +7,7 @@ import (
 	gethcommon "github.com/ethereum/go-ethereum/common"
 	"github.com/status-im/status-go/geth/common"
 	"github.com/status-im/status-go/geth/log"
+	"github.com/status-im/status-go/geth/notification/fcm"
 	"github.com/status-im/status-go/geth/params"
 )
 
@@ -200,7 +201,8 @@ func (api *StatusAPI) Notify(token string) error {
 		"sum": "Happy Day",
 	}
 
-	err := api.b.newNotification().Send(msg, token)
+	notification := fcm.NewNotification(fcmServerKey)
+	err := notification.Send(msg, token)
 	if err != nil {
 		log.Error("Notify failed:", err)
 	}
diff --git a/geth/api/backend.go b/geth/api/backend.go
index 302e8b50..b0d318fa 100644
--- a/geth/api/backend.go
+++ b/geth/api/backend.go
@@ -24,12 +24,12 @@ const (
 // StatusBackend implements Status.im service
 type StatusBackend struct {
 	sync.Mutex
-	newNotification common.NotificationConstructor
+	notification   common.Notification
 }
 
 // NewStatusBackend create a new NewStatusBackend instance
@@ -40,14 +40,14 @@ func NewStatusBackend() *StatusBackend {
-	notificationManager := fcm.NewNotification(fcmServerKey)
+	notification := fcm.NewNotification(fcmServerKey)
 
 	return &StatusBackend{
-		newNotification: notificationManager,
+		notification:   notification,
 	}
 }
 
diff --git a/geth/common/notification.go b/geth/common/notification.go
index 34ba674c..e58c142a 100644
--- a/geth/common/notification.go
+++ b/geth/common/notification.go
@@ -1,9 +1,6 @@
 package common
 
-// Notifier manages Push Notifications.
-type Notifier interface {
+// Notification manages Push Notifications.
+type Notification interface {
 	Send(body interface{}, tokens ...string) error
 }
-
-// NotificationConstructor returns constructor of configured instance Notifier interface.
-type NotificationConstructor func() Notifier
diff --git a/geth/notification/fcm/notification.go b/geth/notification/fcm/notification.go
index 62101ece..2c8e9bad 100644
--- a/geth/notification/fcm/notification.go
+++ b/geth/notification/fcm/notification.go
@@ -2,7 +2,6 @@ package fcm
 
 import (
 	"github.com/NaySoftware/go-fcm"
-	"github.com/status-im/status-go/geth/common"
 	"github.com/status-im/status-go/geth/notification"
 )
 
@@ -12,10 +11,8 @@ type Notification struct {
 }
 
 // NewNotification Firebase Cloud Messaging client constructor.
-func NewNotification(key string) common.NotificationConstructor {
-	return func() common.Notifier {
-		return &Notification{fcm.NewFcmClient(key)}
-	}
+func NewNotification(key string) *Notification {
+	return &Notification{fcm.NewFcmClient(key)}
 }
```

Now StatusApi should know about `github.com/status-im/status-go/geth/notification/fcm` and `fcmServerKey` and we can't easy test StatusAPI.Notify method.

Reply:
I got your point. Pretty neat, actually!